patches/minecraft/net/minecraft/world/entity/TamableAnimal.java.patch

```diff
@@ -0,0 +1,16 @@
+--- a/net/minecraft/world/entity/TamableAnimal.java
++++ b/net/minecraft/world/entity/TamableAnimal.java
+@@ -196,12 +_,10 @@
+       return super.m_7307_(p_21833_);
+    }
+ 
+-   public void m_6667_(DamageSource p_21809_) {
++   public void sendDeathMessage() {
+       if (!this.f_19853_.f_46443_ && this.f_19853_.m_46469_().m_46207_(GameRules.f_46142_) && this.m_142480_() instanceof ServerPlayer) {
+          this.m_142480_().m_6352_(this.m_21231_().m_19293_(), Util.f_137441_);
+       }
+-
+-      super.m_6667_(p_21809_);
+    }
```

Comment:
I'm a bit reluctant on the removal and replacement an override of a method here with a custom method that's called from the superclass. Is there no better option here? I can only think of possibly making an overload for `m_6667_`/`die` which accepts a boolean parameter to skip the `LivingDeathEvent` hook, then calling the hook here ourselves?

Reply:
could we flip the order for the super and the message send code, and check if the entity is dead?

Reply:
That sounds like a better option, please do so.